### PR TITLE
Feature/add wo seasonal to task cache

### PR DESCRIPTION
**Pull request by supermarkion**
From WO summary product definition doc, we have two seasonal summary, and their ranges are: Nov to Mar, and Apr to Oct. It is easy to know their lengths are 5 monthes and 7 monthes. These cases cannot be handled by the existing CLI if we try to cache multi-year task.

Therefore, add two WO seasonal ranges as frequency options to `save-task` module.

The new save-tasks output samples can be found:

* http://dea-public-data-dev.s3-website-ap-southeast-2.amazonaws.com/?prefix=ga_ls_wo_fq_nov_mar_3/dbs/. 
* http://dea-public-data-dev.s3-website-ap-southeast-2.amazonaws.com/?prefix=ga_ls_wo_fq_apr_oct_3/dbs/. 

The new CLI `odc-stats save-tasks --temporal-range=2010-11--P2Y --frequency=nov-mar --grid=au-30 ga_ls_wo_3` output is:`ga_ls_wo_3_2010--P2Y-11--P5M.csv`. Its [content](https://dea-dev-stats-processing.s3-ap-southeast-2.amazonaws.com/wo-seasonal-csv-files/ga_ls_wo_3_2010--P2Y-11--P5M.csv) should be same as `ga_ls_wo_3_2010-11--P5M.csv` [content](https://dea-dev-stats-processing.s3-ap-southeast-2.amazonaws.com/wo-seasonal-csv-files/ga_ls_wo_3_2010-11--P5M.csv) plus `ga_ls_wo_3_2011-11--P5M.csv` [content](https://dea-dev-stats-processing.s3-ap-southeast-2.amazonaws.com/wo-seasonal-csv-files/ga_ls_wo_3_2011-11--P5M.csv). 

The new CLI `odc-stats save-tasks --temporal-range=2010--P2Y --frequency=apr-oct --grid=au-30 ga_ls_wo_3` output is `ga_ls_wo_3_2010--P2Y-04--P7M.csv`. Its [content](https://dea-dev-stats-processing.s3-ap-southeast-2.amazonaws.com/wo-seasonal-csv-files/ga_ls_wo_3_2010--P2Y-04--P7M.csv) should be same as `ga_ls_wo_3_2011-04--P7M.csv` [content](https://dea-dev-stats-processing.s3-ap-southeast-2.amazonaws.com/wo-seasonal-csv-files/ga_ls_wo_3_2011-04--P7M.csv) plus `ga_ls_wo_3_2010-04--P7M.csv` [content](https://dea-dev-stats-processing.s3-ap-southeast-2.amazonaws.com/wo-seasonal-csv-files/ga_ls_wo_3_2010-04--P7M.csv).

### CODE REVIEW

File: libs/stats/odc/stats/utils.py [hunk 1]
```diff
@@ -42,6 +42,44 @@ def bin_generic(
     return tasks
 
 
+def bin_apr_oct(
```
Comment (tebadi) [inline libs/stats/odc/stats/utils.py]: This can be parametrised to accept the months so `bin_nov_mar` can be removed.

Comment (supermarkion) [reply libs/stats/odc/stats/utils.py]: Aha, you are right. I am a little lost when I am coding this. Let me fix it.

Comment (supermarkion) [reply libs/stats/odc/stats/utils.py]: Fix at this [commit](https://github.com/opendatacube/odc-tools/pull/449/commits/d54f457b931290d0dccf3716a3ed498bd2854550).

File: libs/stats/tests/test_utils.py [hunk 2]
```diff
@@ -74,6 +75,48 @@ def verify(task):
...
+    assert binner(datetime(2020, 12, 30)) == ""
+
+    binner = season_binner(wo_nov_to_mar_seasons_rules)
+    assert binner(datetime(2020, 1, 28)) == "2019-11--P5M"
+    assert binner(datetime(2020, 2, 21)) == "2019-11--P5M"
+    assert binner(datetime(2020, 3, 1)) == "2019-11--P5M"
+    assert binner(datetime(2020, 4, 1)) == ""
+    assert binner(datetime(2020, 5, 31)) == ""
+    assert binner(datetime(2020, 10, 31)) == ""
+    assert binner(datetime(2020, 12, 30)) == "2020-11--P5M"
```
Comment (emmaai) [inline libs/stats/tests/test_utils.py:119]: shouldn't it be with `extract_single_season=True`?

Comment (supermarkion) [reply libs/stats/tests/test_utils.py:119]: I also confused for a while. The `extract_single_season` is a switch to control which function to use: `mk_season_rules` or `mk_single_season_rules`. 

The binner here only test the mapping is right or not.

**Review by emmaai — APPROVED** [review]
lgtm

**Pull request merged**